### PR TITLE
Documentation - adding overview section

### PR DESCRIPTION
**Pull request by jjbustamante**
## Motivation

Adding overview section for people coming into kpack project and want to understand a what kpack is about. 

Signed-off-by: Juan Bustamante <jbustamante@vmware.com>

### CODE REVIEW

**Comment by tylerphelan** [comment]
does the `docs/assets/k8s-components.png` asset have some overlapping text?

File: docs/overview.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+## Kubernetes Overview
+
+[Kubernetes](https://kubernetes.io) is a portable, extensible, open-source platform for managing containerized workloads and services. Nowadays, enterprises are upgrading their applications to run on containers, and [Kubernetes](https://kubernetes.io) is the standard platform to run those applications. It provides a framework to run distributed systems resiliently, takes care of scaling and failover of  workloads, and offers capabilities like:
+- Service discovery
+- Storage orchestration
+- Automated rollouts and rollbacks
+- Self-healing
+- Secret and configuration management
+
+The following diagram (from kubernetes [documentation](https://kubernetes.io/docs/concepts/overview/components/)) shows the main components of the [Kubernetes](https://kubernetes.io):
```
Comment (tylerphelan) [inline docs/overview.md:10]: remove `the` ?

**Review by tylerphelan — APPROVED** [review]

File: docs/overview.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+![](/assets/k8s-components.png)
```
[the remark this answers is not in the document]
Comment (jjbustamante) [reply docs/overview.md:12]: 😮 Let me check that!

File: docs/overview.md (hunk 1 above)
Comment (jjbustamante) [reply docs/overview.md]: Done

File: docs/overview.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jjbustamante) [reply docs/overview.md]: Done! yes, I had to remove the / at the beginning of the image path reference

**Review by samj1912 — APPROVED** [review]

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+## Kubernetes Overview
+
+[Kubernetes](https://kubernetes.io) is a portable, extensible, open-source platform for managing containerized workloads and services. Nowadays, enterprises are upgrading their applications to run on containers, and [Kubernetes](https://kubernetes.io) is the standard platform to run those applications. It provides a framework to run distributed systems resiliently, takes care of scaling and failover of  workloads, and offers capabilities like:
```
Comment (matthewmcnew) [inline docs/overview.md:3]:
```suggestion
[Kubernetes](https://kubernetes.io) is a portable, extensible, open-source platform for managing containerized workloads and services. [Kubernetes](https://kubernetes.io) is the standard platform to run those applications. It provides a framework to run distributed systems resiliently, takes care of scaling and failover of  workloads, and offers capabilities like:
```
I think Kubernetes is viable for many types of users. Maybe we can remove the enterprise reference?

**Review by matthewmcnew — APPROVED** [review]

**Review by tomkennedy513 — APPROVED** [review]

**Pull request merged**